node/service/src/lib.rs

```diff
@@ -266,50 +270,67 @@ fn new_partial<RuntimeApi, Executor>(config: &mut Configuration) -> Result<
 	})
 }
 
-fn real_overseer<S: SpawnNamed>(
+fn make_metrics<M: Metrics + Default>(registry: Option<&Registry>) -> Result<M, ServiceError> {
```

Comment:
there is `Metrics::register` convenience function for this

Reply:
also it doesn't propagate the error

Reply:
IMO propagation is important in this case: if the user has provided a promethius registry at all, then all the metrics should appear. That said, I agree that this function works better within `trait Metrics`.

Reply:
:+1: We should remove `Metrics::register` then